### PR TITLE
Fixed `zoomed image won't fit in the pop-up window` for SVG image_format

### PR DESCRIPTION
**Pull request by hummeltech**
When using `image_format = SVG`, zooming in on an image brings up an incorrectly-sized pop-up window.

**Before:**
image

**After:**
image

Addresses [comment](https://github.com/mikaku/Monitorix/issues/132#issuecomment-174805526) from #132

This seems to be caused by [this issue](https://github.com/oetiker/rrdtool-1.x/issues/769)

### CODE REVIEW

**Comment by mikaku** [comment]
@hummeltech,

I've been testing your pull request (Fedora 33, Firefox 91.0.1):

1. When using `SVG`it seems to work for normal-sized images, but I still can see scroll bars in small-sized images (the ones in the right side, in some graphs). Any idea if this can be easily fixed? 
2. When using `PNG` it now appears an over-sized window in all images. This must be fixed.

@Saentist,
> Same bug is also on .PNG images.

I don't see any problem with `PNG` zoomed images.

> Reported this to author few years ago on IRC
> Popup image window is with size of image,
> and image is undersized to fit in it.
> There is a JS code to fix that in few lines.

Yes, I have an entry in my TODO list with a snippet of an IRC talk with you and the only I have is the following code:
```
style="-webkit-user-select: none;cursor: zoom-in;"
```

Please, explain where to put this line in order to fit correctly the zoomed imaged in the pop-up window either for `PNG` and for `SVG`.

**Comment by hummeltech** [comment]
Hello mikaku, I'll take a look into that issue. I will need to test with PNG type as well then, I expected the ternary to work, but it sounds like I will just need to do a bit more testing. I'll push up some changes soon. Thanks!

**Comment by Saentist** [comment]
> @Saentist,
> 
> > Same bug is also on .PNG images.
> 
> I don't see any problem with `PNG` zoomed images.




Is it visible?

**Comment by mikaku** [comment]
> Is it visible?

Yes, it is. I see you are using the Opera browser. Perhaps there are some incompatibilities with JS, as Firefox works well.

Anyway, and in order to support as many browsers as possible, do you know where should be put the line you recommended on IRC?

**Comment by bachandi** [comment]
I noticed the same issue in Safari and Firefox on macOS. 

As this calculation needs to be everywhere maybe a general function in Monitorix.pm could be added for this?
`'width=" . int($picz_width * ($imgfmt_uc == "SVG" ? (4 / 3) : 1) + 0.5) . "`
Could be something like this:
`'width=" . zoomed_width($picz_width, $imgfmt_uc) . "`

**Comment by mikaku** [comment]
> I noticed the same issue in Safari and Firefox on macOS.

Sure, I still waiting some fixes from the author related to my comment https://github.com/mikaku/Monitorix/pull/342#issuecomment-913705303

**Comment by hummeltech** [comment]
Hi @mikaku, I'll take another look at this, I was hoping to see what @Saentist had in mind to address this before continuing. I'll submit some updates soon! Thanks!

**Comment by Saentist** [comment]
> > Is it visible?
> 
> Yes, it is. I see you are using the Opera browser. Perhaps there are some incompatibilities, because Firefox works well.
> 
> Anyway, and in order to support as many browsers as possible, do you know where should be put the line you recommended on IRC?

It work ONLY in Firefox
Chrome Edge Opera Vivaldi (Chromium engine) not working also Internet explorer (latest on W10)

**Comment by hummeltech** [comment]
> > > Is it visible?
> > 
> > 
> > Yes, it is. I see you are using the Opera browser. Perhaps there are some incompatibilities, because Firefox works well.
> > Anyway, and in order to support as many browsers as possible, do you know where should be put the line you recommended on IRC?
> 
> It work ONLY in Firefox Chrome Edge Opera Vivaldi (Chromium engine) not working also Internet explorer (latest on W10)

Aww, that's too bad, thanks for the quick reply!

**Comment by bachandi** [comment]
nvme.pm, amdgpu.pm and nvidiagpu.pm have been added upstream in master. Maybe you could fix them also?

**Comment by mikaku** [comment]
I've applied your last patches but I don't see any new:

Screenshot from 2021-12-21 09-57-31

I'm missing something?
Firefox 95

**Comment by hummeltech** [comment]
Hi @mikaku Mikaku, I am not seeing that behavior on my system. However, I did need to restart the service in order for these changes to take effect, so if you didn't do that, try doing so. I'll look more closely though.

**Comment by mikaku** [comment]
> However, I did need to restart the service in order for these changes to take effect, so if you didn't do that, try doing so.

I'll double check.

BTW thanks for taking care of the code and remove those trailing spaces.

**Comment by hummeltech** [comment]
@mikaku I think I might have figured it out and have applied a fix, previously specifying `image_format = svg` (as opposed to `image_format = SVG`) in the config file would cause this not to work.

**Comment by mikaku** [comment]
> I think I might have figured it out and have applied a fix, previously specifying image_format = svg (as opposed to image_format = SVG) in the config file would cause this not to work.

Ah yes, that probably was the cause.
I'll make some tests with these new patches.

**Comment by hummeltech** [comment]
Thanks @mikaku, I would be happy to add those same tests to run under GitHub Actions if you would like.

File: lib/Monitorix.pm [hunk 1]
```diff
@@ -70,15 +70,25 @@ sub celsius_to {
...
+
+sub picz_js_a_element {
 	my %params = @_;
 
 	my $zoom = (uc($params{config}->{image_format}) eq "SVG") ? (4 / 3) : 1;
 
 	my $picz_width = POSIX::ceil($params{width} * $zoom);
 	my $picz_height = POSIX::ceil($params{height} * $zoom + 0.5);
 
-	return "      <a href=\"javascript:void(window.open('" . $params{config}->{url} . "/" . $params{config}->{imgs_dir} . $params{IMGz} . "','','width=" . $picz_width . ",height=" . $picz_height . ",scrollbars=0,resizable=0'))\"><img src='" . $params{config}->{url} . "/" . $params{config}->{imgs_dir} . $params{IMG} . "' border='0'></a>\n";
```
Comment (mikaku) [inline lib/Monitorix.pm]: The leading spaces here are intended to keep the HTML source code a good indention (where possible).

**Comment by mikaku** [comment]
@hummeltech, now the zoomed images fit perfectly in the pop-up window either with Firefox and Chromium. :smiley: :+1: 

@Saentist, you might want to test this in your Opera to see if it also works.

> I would be happy to add those same tests to run under GitHub Actions if you would like.

Sorry, I'm not familiarized with GitHub Actions, so feel free to do it if you think this might improve your code.

**Comment by mikaku** [comment]
Thanks!

**Comment by Saentist** [comment]
> @hummeltech, now the zoomed images fit perfectly in the pop-up window either with Firefox and Chromium. 😃 👍
> 
> @Saentist, you might want to test this in your Opera to see if it also works.

Please describe how understand that this issue is fixed?

Before "fix" image was zoomed-out to vertical internal size of window.
image
image
must be
image

After "fix" image was zoomed-out to vertical internal size of window, 
but now window is smaller, with make it even smallest.
image
image
must be
image

I guess everyone undarstand fix as: image is 100% size and window is with proper size to fit inside image.

My understand to be fixed issue pop-up not to include image only, but some JS code inside html bracket's, with resize window not image

**Comment by mikaku** [comment]
> Please describe how understand that this issue is fixed?

Is fixed for Firefox and Chromium browsers, that's why I merged it.
I also asked you on Opera, and if it needs more fixes I'm open to merge them as well.

**Pull request merged**